server/test/report.spec.js

```diff
@@ -0,0 +1,80 @@
+import chai, { expect } from 'chai';
+import app from '../../index';
+import report from './stubs/mock-data.report';
+
+chai.use(require('chai-http'));
+
+describe('Article Report routes', () => {
+  // Login to obtain token
+  let token;
+  before(async () => {
+    const loginReponse = await chai.request(app)
+      .post('/api/v1/auth/login')
+      .send({
+        email: 'mofe@okoro.com',
+        password: 'soldier'
+      });
+    const response = loginReponse.body.data;
+    ({ token } = response);
+  });
+  describe('POST /api/v1/reports', () => {
+    it('should return error if comment is not filled', async () => {
+      const res = await chai.request(app)
+        .post('/api/v1/reports')
+        .send(report.invalidReport[0]);
+
+      expect(res).to.have.status(422);
+      expect(res.body).to.have.property('message').eql({ comment: 'comment is required.' });
+    });
+
+    it('should return error if typeId is not filled', async () => {
+      const res = await chai.request(app)
+        .post('/api/v1/reports')
+        .send(report.invalidReport[1]);
+
+      expect(res).to.have.status(422);
+      expect(res.body).to.have.property('message').eql({ typeId: 'typeId is required.' });
+    });
+
+    it('should return error if articleId is not filled', async () => {
+      const res = await chai.request(app)
+        .post('/api/v1/reports')
+        .send(report.invalidReport[2]);
+
+      expect(res).to.have.status(422);
+      expect(res.body).to.have.property('message').eql({ articleId: 'articleId is required.' });
+    });
+
+    it('should return 400 if article does not exist', async () => {
+      const res = await chai.request(app)
+        .post('/api/v1/reports')
+        .set({ Authorization: `Bearer ${token}` })
+        .send(report.invalidReport[3]);
+
+      expect(res).to.have.status(400);
```

Comment:
@cupidcoder nice work,  I think this should be 404 error since the article does not exist

Reply:
Yeah @samorahno about that I just thought that it's more of a bad request since it's a `POST` request and not a `GET`